main.rb

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env ruby
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/MethodLength: Method has too many lines
+
+require_relative './example_school_library_decorator/book'
+require_relative './example_school_library_decorator/person'
+require_relative './example_school_library_decorator/student'
+require_relative './example_school_library_decorator/teacher'
+
+class App
+  def initialize()
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each do |book|
+      print "Title: \"#{book.title}\", Author: #{book.author} \n"
+    end
+  end
+
+  def list_people
+    @people.map do |person|
+      prefix = person.is_a?(Student) ? '[Student]' : '[Teacher]'
+      print "#{prefix} Name: \"#{person.name}\", Id: #{person.id}, Age: #{person.age} \n"
+    end
+  end
+
+  def create_person
+    print "Do you want to create a student (1) or a teacher (2)? [Input the number]: \n"
+    person_type = gets.chomp
+
+    puts 'Age: '
+    age = gets.chomp.to_i
+    puts 'Name: '
+    name = gets.chomp
+    case person_type
+    when '1'
+      puts 'Has parent permission? [Y/N]: '
+      parent_permission = gets.chomp
+      parent_permission = parent_permission.downcase == 'y'
+      @people << Student.new(age, parent_permission, name)
+    when '2'
+      puts 'Specilization: '
+      specialization = gets.chomp
+      @people << Teacher.new(age, specialization, name)
+    end
+    puts 'Person created successfully'
+  end
+
+  def create_book
+    puts 'Title'
+    title = gets.chomp
+    puts 'Author'
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    print "Select a book from the following list by number \n"
+    @books.each_with_index do |book, index|
+      puts "#{index}) ID: #{index}, Title: #{book.title}"
+    end
+    book_index = gets.chomp.to_i
+
+    puts "\nSelect a person from the following list by number (not id)"
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_index = gets.chomp
+
+    puts "\nDate"
+    date = gets.chomp
+    @rentals << Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
+  end
```

Comment:
- [ ] `create_rental` is a well-written method :+1: It displays all the books then it asks to select a book to rent and then it asks for the person which wants to rent it. But after the user inputs all of these it returns the `uninitialized constant` error. This error is classified as `NameError`  and the reason behind this can be that the method refers to a class or module which it can't find. Kindly check it up and fix it.
![Screenshot from 2021-10-15 00-02-16](https://user-images.githubusercontent.com/62705947/137402092-3e74e4dc-fc2d-4137-a820-eea0e8162d27.png)

Reply:
- [ ] Thank you for pointing this out. Fix uninitialized constant error.